MobileClient/src/providers/background-tracker-service.ts

```diff
@@ -68,10 +89,12 @@ export class BackgroundTrackerService {
     let settingsPromise = this.settingsService.get(Settings.BackgroundOptions);
     let tokenPromise = this.settingsService.get(Settings.SecurityToken);
 
-    let allPromises = Promise.all([urlPromise, settingsPromise, tokenPromise]);
+    let allPromises = Promise.all([urlPromise, settingsPromise, tokenPromise, promise]);
 
     allPromises.then((values) => {
-
+      if(!locationStatus){
```

Comment:
Once you resolve the promise with the status value, you should be able to use the values argument to access the status